src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -439,10 +441,48 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 case 1:
                     assert(!instrIsRMW);
-                    if (targetReg != falseReg)
+
+                    if ((targetReg != falseReg))
                     {
-                        GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, falseReg);
+                        // If targetReg is not the same as `falseReg` then need to move
+                        // the `falseReg` to `targetReg`.
+
+                        if (op1->IsTrueAllMask())
+                        {
+                            // We already skip importing ConditionalSelect if op1 == trueAll, however
+                            // if we still see it here, it is because we wrapped the predicated instruction
+                            // inside ConditionalSelect.
+                            // As such, no need to move the `falseReg` to `targetReg`
+                            // because the predicated instruction will eventually set it.
+                            assert(intrin.op3->IsVectorZero());
+                            assert(falseReg == REG_NA);
+                        }
+                        else if (falseReg == REG_NA)
+                        {
+                            // If falseValue is zero, just zero out those lanes of targetReg using `movprfx`
+                            // and /Z
+                            assert(intrin.op3->isContained() && intrin.op3->IsVectorZero());
+                            GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, targetReg, opt);
```

Comment:
what tells the emitter that we need `mask/Z` instead of `mask/M`?

Reply:
`mask/M` if `insScalableOpts == INS_SCALABLE_OPTS_PREDICATE_MERGE` else `mask/Z`